src/main/java/hudson/remoting/URLish.java

```diff
@@ -16,20 +18,28 @@
 abstract class URLish {
     private URLish() {}
 
+    /**
+     * Converts URLish to the standard {@link URL} type.
+     * @return URL or {@code null} if the target destination is known to be non-existent.
```

Comment:
IIUC it is the line

```java
if (u!=null)    return u;
```

that is under discussion. I think it makes sense to have `toURL` throw an exception if it is definitely broken, and for the caller to decide what to do with that—log it, in this case (whatever the `catch` block does).

Or fix this method to be `@Nonnull`, which seems more natural, and leave it up to the consumer of the resulting `URL` to deal with missing resources—they would anyway need to be handling `FileNotFoundException`, so it seems pointless to be prechecking that condition.

IOW lines 46–50 seem stupid.